main.js

```diff
@@ -0,0 +1,114 @@
+let myLibrary = [];
+
+const addbookForm = document.querySelector('#addbook-form');
+const addbookBtn = document.querySelector('#addbook-btn');
+const booksContainer = document.querySelector('#books-list');
+
+addbookForm.style.display = 'none';
+
+function Book(author, title, pages, readStatus) {
+  this.author = author;
+  this.title = title;
+  this.pages = pages;
+  this.readStatus = readStatus;
+  this.node = null;
+  this.removeCallback = null;
+}
+
+Book.prototype.initializeHTML = function initializeHTML() {
+  const bookDiv = document.createElement('div');
+  const titleDiv = document.createElement('div');
+  const authorDiv = document.createElement('div');
+  const pagesDiv = document.createElement('div');
+  const readStatusDiv = document.createElement('div');
+  const removeButton = document.createElement('button');
+  const readStatusToggleButton = document.createElement('button');
+
+  removeButton.innerHTML = 'Remove';
+  readStatusToggleButton.innerHTML = 'Toggle read';
+
+  titleDiv.classList.add('title');
+  authorDiv.classList.add('author');
+  pagesDiv.classList.add('pages');
+  readStatusDiv.classList.add('read-status');
+  removeButton.classList.add('remove');
+  readStatusToggleButton.classList.add('toggle-read-status');
+
+  removeButton.innerHTML = 'Remove';
+
+  bookDiv.appendChild(titleDiv);
+  bookDiv.appendChild(authorDiv);
+  bookDiv.appendChild(pagesDiv);
+  bookDiv.appendChild(readStatusDiv);
+  bookDiv.appendChild(removeButton);
+  bookDiv.appendChild(readStatusToggleButton);
+
+  removeButton.addEventListener('click', () => {
+    bookDiv.parentNode.removeChild(bookDiv);
+    if (this.removeCallback) {
+      this.removeCallback(this);
+    }
+  });
+
+  readStatusToggleButton.addEventListener('click', () => {
+    this.toggleRead();
+    this.updateHTML();
+  });
+
+  this.node = bookDiv;
+};
+
```

Comment:
## Separate DOM manipulation logic from application logic 🧑‍🏭 
- [ ] We need to modularise our codebase by keeping the application logic separated from DOM manipulation tasks ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/dom_logic_separation_requirement.png) from Pathwright).

- [ ] Do this for the rest of the code in this file. 🙏

Reply:
Hey @christianotieno, thanks for the code review.
Can you elaborate on this?
The method you highlighted only contains dom-manipulation code,
Could you maybe provide hints on what we should be doing?

Thanks.

Reply:
- Yes. Kindly let's move them(dom-manipulation functions) to a separate file then call them in our main file which is the application logic.